Sources/SwiftIDEUtils/SyntaxClassifier.swift

```diff
@@ -196,12 +196,37 @@ private struct ClassificationVisitor {
 
     for case (let index, let child?) in children.enumerated() {
 
-      let classification: (SyntaxClassification, Bool)?
+      let classification: (classification: SyntaxClassification, force: Bool)?
       if case .layout(let layout) = descriptor.node.kind.syntaxNodeType.structure {
         classification = SyntaxClassification.classify(layout[index])
       } else {
         classification = nil
       }
+
+      if let classification, classification.force {
+        // Leading trivia.
+        report(range: SyntaxClassifiedRange(kind: .none, range: ByteSourceRange(offset: byteOffset, length: child.leadingTriviaByteLength)))
```

Comment:
We should also classify the trivia for doc comments (both leading and trailing). A test case to test that would be.

```swift
@MyAttribute // some comment
func foo() {}
```

To do that, I think you should factor out the following and call it to classify the trivia.

https://github.com/apple/swift-syntax/blob/e04c5c117bdf5a92ddc309ac4ce8e33fe302db41/Sources/SwiftIDEUtils/SyntaxClassifier.swift#L168-L173

Reply:
Ah, yeah. But we also have to add two public properties `leadingTriviaPieces` and `trailingTriviaPieces` in `RawSyntax` to get the trivia pieces as in my new commit.

Reply:
Oh, I thought we could have a function to classify the trivia pieces so we don’t need to repeat this loop everywhere. 

Something like

```swift
/// Classifies `trivia` and returns the number of bytes the trivia took up in the source
func classify(trivia: [RawTriviaPiece]) -> Int {
  var classifiedBytes = 0
  for triviaPiece in triviaPieces {
    let range = triviaPiece.classify(offset: byteOffset)
    report(range: range)
    classifiedBytes += triviaPiece.byteLength
  }
  return classifiedBytes
}
```

And then here you can just do

```swift
if let triviaPieces = child.leadingTriviaPieces {
  byteOffset += classify(trivia: triviaPieces)
}
```

And you can also use the function in `handleToken`.

My thinking is that repeating an implementation twice is still OK but as soon as we reach 3 or more copies, we should really factor it out.

Reply:
Ah, I think that looks nice. 👍